core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -406,6 +407,19 @@ public QueryManagerConfig setQueryMaxScanPhysicalBytes(DataSize queryMaxScanPhys
         return this;
     }
 
+    @Min(1)
+    public int getQueryReportedRuleStatsLimit()
+    {
+        return queryReportedRuleStatsLimit;
+    }
+
+    @Config("query.reported-rule-stats-limit")
```

Comment:
TBH, I would probably keep it as static

Reply:
I've followed your comment from the other PR. In some problematic cases, this parameter could easily give us more rule statistics than the default 20 without the need to deploy a new version of Trino. But it is up to you if you want to remove it.